shared/src/consensus/mod.rs

```diff
@@ -13,10 +14,25 @@ pub struct ConsensusFault {
     pub fault_type: ConsensusFaultType,
 }
 
+/// Result of checking two headers for a consensus fault, with the gas used
+/// Only for v14 and earlier
+#[derive(Clone, Debug)]
+pub struct ConsensusFaultWithGas {
+    /// Address of the miner at fault (always an ID address).
+    pub target: ActorID,
+    /// Epoch of the fault, which is the higher epoch of the two blocks causing it.
+    pub epoch: ChainEpoch,
+    /// Type of fault.
+    pub fault_type: ConsensusFaultType,
+    /// Gas used in checking the fault
+    pub gas_used: i64,
+}
+
 /// Consensus fault types in VM.
 #[derive(FromPrimitive, Clone, Copy, Debug)]
 #[repr(u8)]
 pub enum ConsensusFaultType {
+    NoFault = 0,
```

Comment:
I'm pretty strongly against a "no fault" variant as it means all the other fields in the `ConsensusFault` object will be garbage. We only do this kind of thing in go because go doesn't have enums/options.

Reply:
synced on this: decision is to move the "no fault" (Go) -> Option (Rust) step into the FFI